CMakeLists.txt

```diff
@@ -115,21 +113,19 @@ add_library(
 )
 
 if(${TRITON_ENABLE_GPU})
-  set(HOST_COMPILER_FLAGS "")
-  if (WIN32)
-    set(HOST_COMPILER_FLAGS "/MD")
-  else()
-    set(HOST_COMPILER_FLAGS "-fPIC")
-  endif()
-
-  set(CUDA_LIBRARIES PUBLIC ${CUDA_LIBRARIES})
-  cuda_add_library(
+  add_library(
     kernel_library_new
     src/kernel.cu src/kernel.h
-    OPTIONS -arch compute_53
-    OPTIONS -code compute_53,sm_53,sm_60,sm_61,sm_62,sm_70,sm_72,sm_75
-    OPTIONS -Xcompiler ${HOST_COMPILER_FLAGS}
   )
+
+  enable_language(CUDA)
+  set_target_properties(kernel_library_new PROPERTIES LANGUAGE CUDA)
+  set_target_properties(kernel_library_new PROPERTIES CUDA_ARCHITECTURES "53;60;61;62;70;72;75")
```

Comment:
Just for my own curiosity - any idea what the implications of this line not including Ampere+ (>=8.0)?

Does it mean this will be JIT'd at startup, or just unsupported?

Reply:
That's a great question that I don't know the answer to.

Reply:
Not sure what is the new variable name for it is, `code` defines the optimized kernel for the listed CC, `arch` defines the PTX generated which provides support for CC not listed above.